src/App.tsx

```diff
@@ -1,9 +1,28 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
 export const App: React.FC = () => {
+  const [query, setQuery] = useState('');
+  const [visibleMovies, setVisibleMovies] = useState(moviesFromServer);
+
+  const handleInput: React.ChangeEventHandler<HTMLInputElement> = event => {
+    const userInput = event.target.value;
+
+    setQuery(userInput);
+
+    const filtered = moviesFromServer.filter(
+      movie =>
+        movie.title.toLowerCase().includes(userInput.toLowerCase().trim()) ||
+        movie.description
+          .toLowerCase()
+          .includes(userInput.toLowerCase().trim()),
+    );
+
+    setVisibleMovies(filtered);
+  };
```

Comment:
```suggestion

  const handleInput: React.ChangeEventHandler<HTMLInputElement> = event => {
    const userInput = event.target.value;

    setQuery(userInput); 
  };
  
    const visibleMovies = moviesFromServer.filter(
      movie =>
        movie.title.toLowerCase().includes(query.toLowerCase().trim()) ||
        movie.description
          .toLowerCase()
          .includes(query.toLowerCase().trim()),
    );
```
you don't need that state at all